docs/general/developer-resources/emerald-paratime/writing-dapps-on-emerald.md

```diff
@@ -4,39 +4,173 @@ description: How to write and deploy a dApp to Oasis Emerald
 
 # Writing dApps on Emerald
 
-This tutorial will show you how to set up a dApp development environment using Hardhat and Truffle deployment tools or the Remix IDE. Oasis Emerald exposes an EVM-compatible interface, so writing dApps isn't much different compared to the original Ethereum network. However, there are still some important differences which this guide will walk you through.
+This tutorial will show you how to set up dApp development environment for
+Emerald. We will walk you through the Hardhat and Truffle configuration and -
+for those who prefer a simpler web-only interface - the Remix IDE. Oasis
+Emerald exposes an EVM-compatible interface, so writing dApps isn't much
+different compared to the original Ethereum network!
+
+## Oasis Consensus Layer and Emerald ParaTime
+
+Oasis network consists of the Consensus layer and a number of Layer 2 chains
+called the ParaTimes (to learn more, check the [Oasis Network Overview]
+[overview] chapter). Emerald is a ParaTime which implements the Ethereum
+Virtual Machine.
+
+The minimum and also expected block time in Emerald is **6 seconds**. Any
+Emerald transaction will require at least this amount of time to be executed.
+
+The native Oasis addresses are bech32-encoded (e.g. `oasis1qpupfu7e2n6pkezeaw0yhj8mcem8anj64ytrayne`)
+while Emerald supports both the bech32-encoded and the Ethereum-compatible hex-
+encoded addresses (e.g. `0x90adE3B7065fa715c7a150313877dF1d33e777D5`). The
```

Comment:
Put `hex-encoded` on a single line so it doesn't create a space in the final rendered HTML.